### PR TITLE
improving db-checkbook tests

### LINKED ISSUE #117: db-checkbook test coverage limited, some tests skipping
tagging @DeaBardhoshi 

some unit tests from the original implementation of the db-checkbook pipeline are being skipped, and in general the test coverage is currently limited to verifying the cleaning and grouping process for `checkbook` and `CPDB` input datasets. Two main goals for this issue:

- [x] Refactor `build.py` functions to allow for easier testing so that tests currently being skipped in the product run successfully. 
- [x] Ensure that existing unit tests are useful and doing what we'd like them to do. 

And a nice-to-have:
- [x] Identify other areas of db-checkbook build that should be tested to ensure the quality of the data, then write these tests.

### PR DESCRIPTION
**Pull request by athursland**
Fixes #117 - tagging @DeaBardhoshi for thoughts

### changes
- `db-checkbook/sql_query/query.sql` which we use for the Checkbook category assignment based on `Budget Code` and `Contract Purpose` was missing the **_entire_** Fixed Asset query, which was revealed by testing. Added this query to the top of the file. 
- Organized `test_build_output.py` file into three pytest classes, each addressing a specific step in the transformation step: `TestCheckbook`: validating the results of cleaning and grouping the Checkbook input data, `TestCPDB`:  validating the results of merging the various versions of CPDB geometries, and `TestHistoricalLiquidations`: validating the results of category assignment on the final merged dataset. Overall refactored this file to be less repetitive, verified usefulness of tests.
- Removed default arguments from `build.py` fns - they now expect to be passed a pandas or geopandas df as an argument. Previously, we'd added default arguments for testing, but these proved to be a hassle and introduced their own problems. 
- Separated reading input data from cleaning input data in `build.py`
- Fixed a couple of typos in `generate_test_data.py`.

### CODE REVIEW

File: products/db-checkbook/test/test_build_output.py [hunk 1]
```diff
@@ -108,7 +133,51 @@ def test_high_sensitivity_fixed_asset():
...
+    def test_final_category(self):
+        assert self.historical_liquidations["final_category"].equals(
+            self.expected_historical_liquidations["final_category"]
+        )
+
+    def test_all_categories(self):
+        cols = ["cpdb_category", "bc_category", "cp_category", "final_category"]
+        assert self.historical_liquidations[cols].equals(
+            self.expected_historical_liquidations[cols]
+        )
```
Comment (damonmcc) [inline products/db-checkbook/test/test_build_output.py:178]: does this one cover the 4 column-specific tests above?

File: products/db-checkbook/build_scripts/build.py [hunk 2]
```diff
@@ -88,7 +85,7 @@ def _group_checkbook(data: pd.DataFrame = None) -> pd.DataFrame:
     """
     :return: checkbook nyc data grouped by capital project
     """
-    if not data:
+    if data.empty:
```
Comment (fvankrieken) [inline products/db-checkbook/build_scripts/build.py:88]: Why this check? In what case will empty data be fed in?

File: products/db-checkbook/sql_query/query.sql [hunk 3]
```diff
@@ -1,3 +1,303 @@
+UPDATE capital_projects SET col_category = 'Fixed Asset'
```
Comment (fvankrieken) [inline products/db-checkbook/sql_query/query.sql:1]: sql is a more apt name for this folder, and the file itself could use a more descriptive name

File: products/db-checkbook/build_scripts/build.py (hunk 2 above)
Comment (athursland) [reply products/db-checkbook/build_scripts/build.py:88]: There was a version of testing where I added default params for all the build fns in order to pass in test data but ended up scrapping it. Missed this one, will remove

File: products/db-checkbook/test/test_build_output.py (hunk 1 above)
Comment (damonmcc) [reply products/db-checkbook/test/test_build_output.py:178]: gotcha. using `assert_frame_equal` in the all category test _might_ give results informative enough to isolate a problem with a specific category, but feels like a nice-to-have thing to maybe try later

File: products/db-checkbook/build_scripts/build.py (hunk 2 above)
Comment (damonmcc) [reply products/db-checkbook/build_scripts/build.py:88]: still wanted to drop this?

**Review by damonmcc — APPROVED** [review]

**Pull request merged**